Cabal/Distribution/Package.hs

```diff
@@ -303,7 +313,20 @@ instance Text Dependency where
              Parse.skipSpaces
              return (Dependency name ver)
 
+instance Text ExeDependency where
+  disp (ExeDependency name exe ver) =
+    (disp name <<>> Disp.text ":" <<>> Disp.text exe) <+> disp ver
+
+  parse = do name <- parse
+             _ <- Parse.char ':'
+             exe <- Parse.munch1 isAlphaNum
```

Comment:
I think the executable name format is more flexible than AlphaNum. Safest is to parse as if it were a PackageName.